src/presentation/api/controller/ssl.go

```diff
@@ -0,0 +1,185 @@
+	return &SslController{
+		persistentDbSvc: persistentDbSvc,
+		transientDbSvc:  transientDbSvc,
+		sslService:      service.NewSslService(persistentDbSvc, transientDbSvc),
+	}
+}
+
+// ReadSslPairs	 godoc
+// @Summary      ReadSslPairs
+// @Description  List ssl pairs.
+// @Tags         ssl
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Success      200 {array} entity.SslPair
+// @Router       /v1/ssl/ [get]
+func (controller *SslController) Read(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(c, controller.sslService.Read())
+}
+
+func parseRawVhosts(rawVhostsInput interface{}) (rawVhosts []string, err error) {
+	rawVhostsSlice, assertOk := rawVhostsInput.([]interface{})
+	if !assertOk {
+		rawVhostUniqueStr, err := voHelper.InterfaceToString(rawVhostsInput)
+		if err != nil {
+			return rawVhosts, errors.New("VirtualHostsMustBeStringOrStringSlice")
+		}
+		return append(rawVhosts, rawVhostUniqueStr), err
+	}
+
+	rawVhosts = []string{}
+	for _, rawVhost := range rawVhostsSlice {
+		rawVhostStr, err := voHelper.InterfaceToString(rawVhost)
+		if err != nil {
+			slog.Debug(err.Error(), slog.Any("vhost", rawVhost))
+			continue
+		}
+		rawVhosts = append(rawVhosts, rawVhostStr)
+	}
+
+	return rawVhosts, err
+}
+
+// CreateSslPair    	 godoc
+// @Summary      CreateSslPair
+// @Description  Create a new ssl pair.
+// @Tags         ssl
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Param        createSslPairDto 	  body    dto.CreateSslPair  true  "All props are required.<br />virtualHosts may be string or []string. Alias is not allowed.<br />certificate is a string field, i.e. ignore the structure shown.<br />certificate and key must be base64 encoded."
+// @Success      201 {object} object{} "SslPairCreated"
+// @Router       /v1/ssl/ [post]
+func (controller *SslController) Create(c echo.Context) error {
+	requestBody, err := apiHelper.ReadRequestBody(c)
+	if err != nil {
+		return err
+	}
+
+	rawVhosts, err := parseRawVhosts(requestBody["virtualHosts"])
+	if err != nil {
+		return apiHelper.ResponseWrapper(c, http.StatusBadRequest, err.Error())
+	}
+	requestBody["virtualHosts"] = rawVhosts
+
+	encodedCert, err := valueObject.NewEncodedContent(requestBody["certificate"])
+	if err != nil {
+		return apiHelper.ResponseWrapper(c, http.StatusBadRequest, err.Error())
+	}
+	decodedCert, err := encodedCert.GetDecodedContent()
+	if err != nil {
+		return apiHelper.ResponseWrapper(
+			c, http.StatusBadRequest, "CannotDecodeSslCertificateContent",
+		)
+	}
+	requestBody["certificate"] = decodedCert
+
+	encodedKey, err := valueObject.NewEncodedContent(requestBody["key"])
+	if err != nil {
+		return apiHelper.ResponseWrapper(c, http.StatusBadRequest, err.Error())
+	}
+	decodedKey, err := encodedKey.GetDecodedContent()
+	if err != nil {
+		return apiHelper.ResponseWrapper(
+			c, http.StatusBadRequest, "CannotDecodeSslPrivateKeyContent",
+		)
+	}
+	requestBody["key"] = decodedKey
```

Comment:
Same thing here.

Reply:
What I mentioned about the certificate content also applies here.